### PR TITLE
Add tests and their results as return values from run API calls

### PR DESCRIPTION
**Pull request by rmeritz**
See jsdoc for more information on the spec followed. It was largely cribbed from the existing `getTestsForRunsForCycle` cycle service

### CODE REVIEW

File: server/services/RunService.js [hunk 1]
```diff
@@ -393,7 +410,27 @@ async function sequelizeRunsToJsonRuns(sequelizeRuns) {
             run_status_id: run.run_status_id,
             run_status: run.RunStatus.name,
             test_version_id: run.test_version_id,
-            testers: run.Users.map(u => u.id)
+            testers: run.Users.map(u => u.id),
+            tests: run.ApgExample.Tests.reduce((acc, test) => {
```
Comment (spectranaut) [inline server/services/RunService.js:414]: getting the tests for a run is a little more complicated that this -- you need to use TestToAt to make sure that any test applies to a run with a specific AT. Not all test files for an APG pattern apply to all ATs.

For example, of all Checkbox APG example tests, 2 of them do not apply to VoiceOver, so those test need to be skipped in all the runs of VoiceOver.

File: server/services/RunService.js [hunk 2]
```diff
@@ -393,7 +410,27 @@ async function sequelizeRunsToJsonRuns(sequelizeRuns) {
...
+                acc.push({
+                    id: test.id,
+                    file: test.file,
+                    name: test.name,
+                    execution_order: test.execution_order,
+                    results: test.TestResults.filter(
+                        testResult => testResult.run_id == run.id
```
Comment (spectranaut) [inline server/services/RunService.js:421]: this might take care for that for results? but you still will see a test with no results.

**Review by spectranaut — CHANGES_REQUESTED** [review]
One more detail of the tests needs to be accounted for! Otherwise looking really good.

**Review by evmiguel — COMMENTED** [review]
Reviewed, waiting for Val's requests to be fulfilled.

**Comment by rmeritz** [comment]
@spectranaut || @evmiguel - I fixed the issue with non-relavant tests being included in the results. See the last commit.

File: server/services/RunService.js [hunk 3]
```diff
@@ -393,7 +410,31 @@ async function sequelizeRunsToJsonRuns(sequelizeRuns) {
             run_status_id: run.run_status_id,
             run_status: run.RunStatus.name,
             test_version_id: run.test_version_id,
-            testers: run.Users.map(u => u.id)
+            testers: run.Users.map(u => u.id),
+            tests: run.ApgExample.Tests.filter(test =>
+                (test.TestToAts || []).some(
```
Comment (evmiguel) [inline server/services/RunService.js:415]: This change should handle the change request @spectranaut

**Review by evmiguel — APPROVED** [review]

**Pull request merged**